### PR TITLE
`make format` improvements

### PR DESCRIPTION
**Pull request by mithro**
* Update to `clang-format-5.0`
 * Use `yapf` from the environment if it exists.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -15,11 +18,13 @@ build:
...
+FORMAT_EXCLUDE = third_party git env build
+FIND_EXCLUDE = $(foreach x,$(FORMAT_EXCLUDE),-and -not -path './$(x)/*')
 format:
-	find . -name \*.cc -and -not -path './third_party/*' -and -not -path './.git/*' -print0 | xargs -0 -P $$(nproc) ${CLANG_FORMAT} -style=file -i
-	find . -name \*.h -and -not -path './third_party/*' -and -not -path './.git/*' -print0 | xargs -0 -P $$(nproc) ${CLANG_FORMAT} -style=file -i
-	find . -name \*.py -and -not -path './third_party/*' -and -not -path './.git/*' -print0 | xargs -0 -P $$(nproc) yapf -p -i
-	find . -name \*.tcl -and -not -path './third_party/*' -and -not -path './.git/*' -print0 | xargs -0 -P $$(nproc) -n 1 ${XRAY_TCL_REFORMAT} 2>/dev/null
+	find . -name \*.cc $(FIND_EXCLUDE) -print0 | xargs -0 -P $$(nproc) ${CLANG_FORMAT} -style=file -i
+	find . -name \*.h $(FIND_EXCLUDE) -print0 | xargs -0 -P $$(nproc) ${CLANG_FORMAT} -style=file -i
+	$(IN_ENV) find . -name \*.py $(FIND_EXCLUDE) -print0 | xargs -0 -P $$(nproc) yapf -p -i
```
Comment (JohnDMcMaster) [inline Makefile]: Is this correct? At a quick glance this looks to me like find is executed in the environment instead of yapf

File: .travis.yml [hunk 2]
```diff
@@ -27,8 +27,9 @@ matrix:
         apt:
           sources:
             - ubuntu-toolchain-r-test
+            - llvm-toolchain-trusty-5.0
           packages:
-            - clang-format-3.9
+            - clang-format-5.0
```
Comment (JohnDMcMaster) [inline .travis.yml]: Out of curiosity why did you chose version 5.0 specifically? When .travis and CLANG_FORMAT got out of sync before I had issues with the newer version running on travis, which is why I rolled back to 3.9. But it seems like it accepted this, so guess its fine

Comment (mithro) [reply .travis.yml]: clang-format-5.0 seems the latest available on both our local machines and on travis.

File: Makefile (hunk 1 above)
Comment (mithro) [reply Makefile]: Environment is inherit through the find into yapf. I have confirmed this works locally.

**Pull request merged**